### PR TITLE
add docs

### PR DESCRIPTION
**Pull request by inftord**
:3

### CODE REVIEW

File: .github/docs.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+jobs:
+  build:
+    runs-on: self-hosted
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 7.0.103
```
Comment (OoLunar) [inline .github/docs.yml]: We use .NET 8, this might break.

File: .gitignore [hunk 2]
```diff
@@ -483,4 +483,5 @@ $RECYCLE.BIN/
 build/
 logs/
 res/config.debug.json
-benchmark-results.md
\ No newline at end of file
+benchmark-results.md
+docs/_site/
```
Comment (OoLunar) [inline .gitignore:487]: This was already ignored, I think?

**Review by OoLunar — APPROVED** [review]
LGTM. Thank you very much!

File: .github/docs.yml (hunk 1 above)
Comment (inftord) [reply .github/docs.yml]: ...can i use preview versions with setup-dotnet?

File: .gitignore (hunk 2 above)
Comment (inftord) [reply .gitignore:487]: nope

File: .github/docs.yml (hunk 1 above)
Comment (OoLunar) [reply .github/docs.yml]: https://github.com/OoLunar/HyperSharp/blob/e9823cc436e11998a598f13b08017c09025631e6/.github/workflows/build-commit.yml#L29-L32

**Comment by OoLunar** [comment]
LGTM. Thank you!

**Pull request merged**